src/value.rs

```diff
@@ -64,33 +59,34 @@ impl Value {
           None
         }
       },
-      &Value::Array(ref vec) => {
-        if pointer.len() == 0 {
+      Value::Array(ref vec) => {
+        if pointer.is_empty() {
           Some(self.clone())
         } else if let Some(value) = pointer.remove(0).parse::<usize>().ok().map_or(None, |i| vec.get(i)) {
           value.get(pointer)
         } else {
           None
         }
-      }
+      },
+      _ => if pointer.is_empty() { Some(self.clone()) } else { None }
     }
   }
 
   /// Converts a value into a JSON string for distribution.
-  pub fn to_json(self) -> String {
+  pub fn into_json(self) -> String {
```

Comment:
Really don't like this method rename. We aren't converting to a json type, we are converting to a json string format.